docs/design/proposals/multi-version/multi-version.md

```diff
@@ -0,0 +1,259 @@
+  and generating mutating webhooks in ACK.
+- Provide a high-level plan for how to implement multi-versioning and
+  generating mutating webhooks in ACK
+- Discuss best practices for testing and validation of the new features
+
+
+## Non-Goals
+
+- Provide a comprehensive overview of all features of ACK
+- Provide detailed instructions for using ACK
+- Discuss implementation details outside of the scope of multi-versioning
+and generating mutating webhooks in ACK
+
+## Prior work
+
+Prior to this design, several components were built to support
+multi-versioning in Amazon Controllers for Kubernetes (ACK). The focus
+of these efforts was mainly on modifying the ACK/code-generator tool
+to generate API definitions for different API versions and enabling
+the webhook server through controller flags. A key contribution was the
+generalization of webhook generation beyond just mutating webhooks,
+as described in the webhook PR.
+
+Github [issue][mv-gh-issue] and 
+code-generator PRs:
+- https://github.com/aws-controllers-k8s/code-generator/pull/108
+- https://github.com/aws-controllers-k8s/code-generator/pull/111
+- https://github.com/aws-controllers-k8s/code-generator/pull/121
+- https://github.com/aws-controllers-k8s/code-generator/pull/122
+
+### Issue with prior work
+
+However, one issue that prevented completion of the webhook PR was the
+difference between how ACK views CRD versions and how they are handled
+in Kubernetes. In ACK, all resources are versioned with the same APIVersion,
+while in Kubernetes, resources can have multiple versions and can deprecate
+at various stages and levels. This makes it challenging to implement
+webhooks that can handle multiple versions and deprecations effectively.
+
+## Overview of the design
+
+The proposed design aims to address the challenges of multi-versioning
+and generating mutating webhooks in Amazon Controllers for Kubernetes (ACK).
+It involves modifications to the ACK/code-generator tool and the implementation
+of a webhook server to handle multiple versions and deprecations of Custom
+Resource Definitions (CRDs). The solution uses a combination of Kubernetes API
+machinery and controller-runtime libraries, and the document outlines a
+high-level plan for implementation, including updating the code-generator
+tool and modifying ACK's controllers. Additionally, the document discusses
+the testing strategies that will be employed to ensure backward compatibility.
+
+### Rethinking the ACK versioning
+
+Before jumping to any special topic, we need to answer two important questions:
+- Do we still want to bump all the CRD versions all at once? meaning that
+  if we rename a field in CRD A, CRD B C D and E will also need a version bump
+  (and a mutating webhook handler) even if they are not modified?
+- How do we deprecate/remove versions in a programatical way?
+
+#### CRDs Mono Version vs Group Version
+
+The answer to the questions above will impact the design of [`metadata.yaml`]
```

Comment:
State explicitly here that `metadata.yaml` is where API versions are set for ACK controllers.